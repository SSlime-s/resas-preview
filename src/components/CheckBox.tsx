interface Props extends React.InputHTMLAttributes<HTMLInputElement> {
	label: string;
}
export function CheckBox({ label, ...props }: Props) {
	return (
		<label>
			<input type="checkbox" {...props} />
			{label}
		</label>
	);
}
