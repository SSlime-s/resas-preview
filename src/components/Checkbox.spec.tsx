import { render } from "@testing-library/react";

import { CheckBox } from "./CheckBox";

function setup(checked: boolean) {
	const result = render(<CheckBox checked={checked} label="hoge" />);

	return { ...result };
}

describe("Components/CheckBox", () => {
	it("スナップショット: unchecked", () => {
		const { container } = setup(false);

		expect(container).toMatchSnapshot();
	});

	it("スナップショット: checked", () => {
		const { container } = setup(true);

		expect(container).toMatchSnapshot();
	});
});
