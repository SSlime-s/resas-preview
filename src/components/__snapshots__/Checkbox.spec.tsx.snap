// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components/CheckBox > スナップショット: checked 1`] = `
<div>
  <label
    class="🐻-3595835307"
  >
    <input
      checked=""
      class="🐻-3042886335"
      type="checkbox"
    />
    <span
      class="🐻-3332179769"
    >
      <svg
        fill="currentColor"
        height="1.25rem"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1.25rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M19 3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2zm-9 14-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </span>
    <span
      class="🐻-4135248302"
    >
      hoge
    </span>
  </label>
</div>
`;

exports[`Components/CheckBox > スナップショット: unchecked 1`] = `
<div>
  <label
    class="🐻-3595835307"
  >
    <input
      class="🐻-3042886335"
      type="checkbox"
    />
    <span
      class="🐻-3332179769"
    >
      <svg
        fill="currentColor"
        height="1.25rem"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1.25rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
    <span
      class="🐻-4135248302"
    >
      hoge
    </span>
  </label>
</div>
`;
